credentials/handler.go

```diff
@@ -0,0 +1,267 @@
+			i, _, err := h.r.PrivilegedIdentityPool().FindByCredentialsIdentifier(ctx, identity.CredentialsTypeOIDC, uid(provider))
+			if err != nil {
+				return nil, err
+			}
+			if i != nil {
+				return nil, errors.New("These OIDC credentials are already used by another identity")
+			}
+		}
+
+		var b bytes.Buffer
+		if err := json.NewEncoder(&b).Encode(newConfig); err != nil {
+			return nil, err
+		}
+		return &identity.Credentials{
+			Type:        newCreds.Type,
+			Identifiers: []string{},
+			Config:      b.Bytes(),
+		}, nil
+	} else if newCreds.Type == identity.CredentialsTypePassword {
+		var newConfig password.CredentialsConfig
+		if err := json.Unmarshal(newCreds.Config, &newConfig); err != nil {
+			return nil, err
+		}
+		var b bytes.Buffer
+		if err := json.NewEncoder(&b).Encode(newConfig); err != nil {
+			return nil, err
+		}
+		return &identity.Credentials{
+			Type:        newCreds.Type,
+			Identifiers: []string{},
+			Config:      b.Bytes(),
+		}, nil
+	} else {
+		return nil, errors.New("Illegal credentials type")
+	}
+}
+
+func (h *Handler) UpdateCreds(ctx context.Context, identityId uuid.UUID, oldCreds *identity.Credentials, newCreds Body) error {
+	if newCreds.Type == identity.CredentialsTypeOIDC {
+		// for OIDC - replace the ones in the import and leave all the rest w/o changing
+		var newConfig oidc.CredentialsConfig
+		if err := json.Unmarshal(newCreds.Config, &newConfig); err != nil {
+			return err
+		}
+		var oldConfig oidc.CredentialsConfig
+		if err := json.Unmarshal(oldCreds.Config, &oldConfig); err != nil {
+			return err
+		}
+		for _, provider := range newConfig.Providers {
+			i, _, err := h.r.PrivilegedIdentityPool().FindByCredentialsIdentifier(ctx, identity.CredentialsTypeOIDC, uid(provider))
+			if err != nil {
+				return err
+			}
+			if i != nil && i.ID != identityId {
+				return errors.New("These OIDC credentials are already used by another identity")
+			}
+		}
+
+		var newProviders []oidc.ProviderCredentialsConfig
+		for _, oldV := range oldConfig.Providers {
+			added := false
+			for _, newV := range newConfig.Providers {
+				if newV.Provider == oldV.Provider {
+					h.r.Logger().
+						WithField("new", newV).
+						WithField("oldV", oldV).
+						Info("provider match")
+					// new old matches old new - replace it
+					newProviders = append(newProviders, newV)
+					added = true
+					break
+				}
+			}
+			if !added {
+				h.r.Logger().
+					WithField("oldV", oldV).
+					Info("no matches")
+				// no matches - keep the old one
+				newProviders = append(newProviders, oldV)
+			}
+		}
+		// add all the new ones that haven't been added
+		for _, newV := range newConfig.Providers {
+			added := false
+			for _, oldV := range oldConfig.Providers {
+				if newV.Provider == oldV.Provider {
+					added = true
+					break
+				}
+			}
+			if !added {
+				// no matches - add a new one
+				newProviders = append(newProviders, newV)
+			}
+		}
+
+		oldConfig.Providers = newProviders
```

Comment:
I think this will replace old links with only new ones(ommiting the ones that are already in the database)

Reply:
sorry my bad I missed line 224